tests/kernel/mutex/sys_mutex/src/thread_competition.c

```diff
@@ -48,9 +45,8 @@ static void high_prio_t1_wait_for_mutex(void *p1, void *p2, void *p3)
 
 	sys_mutex_lock(pmutex, K_FOREVER);
 
-	flag[index] = HIGH_T1;
+	flag[0] = HIGH_T1;
```

Comment:
with this change, when you invoke assert to check flag[0] later, this only guarantees that this thread handler function got executed, it can't guarantee it was the first one to be executed.
the goal of this test, when we unlock a mutex, it should be given to the highest priority and longest waiting time in the mutex's waiting queue.